build_runner/lib/src/generate/phase.dart

```diff
@@ -63,5 +64,28 @@ class BuildAction {
   }
 
   @override
-  String toString() => '$builder on $inputSet';
+  String toString() {
+    final settings = <String>[];
+    if (isOptional) settings.add('optional');
+    if (hideOutput) settings.add('hidden');
+    var result = '$builder on $inputSet';
+    if (settings.isNotEmpty) result += ' $settings';
+    return result;
+  }
+
+  @override
+  bool operator ==(Object other) =>
```

Comment:
This should be doing some sort of check for the builder as well - same for the hashCode. Previously this was just checking `runtimeType.toString().hashCode`, and I would like to maintain at least that much.

Reply:
ugh, how did I miss that!